libV2/process.js

```diff
@@ -0,0 +1,40 @@
+const { EventEmitter } = require('events');
+const os = require('os');
+const { logger } = require('./utils');
+
+
+class Process extends EventEmitter {
+    async setup() {
+        const cleanUpFunc = this.join.bind(this);
+        process.on('SIGINT', cleanUpFunc);
+        process.on('SIGHUP', cleanUpFunc);
+        process.on('SIGQUIT', cleanUpFunc);
+        process.on('SIGTERM', cleanUpFunc);
```

Comment:
A loop?
```suggestion
        ['SIGINT', 'SIGHUP', 'SIGQUIT', 'SIGTERM'].forEach(eventName => {
            process.on(eventName, cleanUpFunc);
        };
```

Although, I am not sure if this will mess up what `this` refers to inside the cleanup function. So this may or may not work...